sormas-backend/src/main/java/de/symeda/sormas/backend/externaljournal/PatientDiaryClient.java

```diff
@@ -211,31 +213,47 @@ public PatientDiaryResult deletePatientDiaryPerson(String personUuid, Runnable c
 	private Invocation.Builder getExternalDataPersonInvocationBuilder(String personUuid) {
 		String externalDataUrl = configFacade.getPatientDiaryConfig().getProbandsUrl() + "/external-data/" + personUuid;
 		Client client = ClientHelper.newBuilderWithProxy().build();
-		return client.target(externalDataUrl).request(MediaType.APPLICATION_JSON).header("x-access-token", getPatientDiaryAuthToken());
+		return client.target(externalDataUrl).request(MediaType.APPLICATION_JSON).header("x-access-token", getPatientDiaryAuthToken(false));
 	}
 
 	/**
 	 * Retrieves a token used for authenticating in the patient diary. The token will be cached.
-	 *
+	 * 
+	 * @param frontendRequest
+	 *            if true && a interface.patientdiary.frontendAuthurl is configured, this url is used to fetch the token.
+	 *            Otherwise, the interface.patientdiary.authurl is used.
 	 * @return the authentication token
 	 */
-	public String getPatientDiaryAuthToken() {
+	public String getPatientDiaryAuthToken(boolean frontendRequest) {
 		try {
-			return authTokenCache.get(PATIENT_DIARY_KEY, this::getPatientDiaryAuthTokenInternal);
+			if (frontendRequest && StringUtils.isNotBlank(configFacade.getPatientDiaryConfig().getFrontendAuthUrl())) {
```

Comment:
if  the request is made with `frontendRequest` set to `true`, but the `frontendAuthUrl` is not configured, this will fallback to use the `backendAuthToken`. Is this intended or should the call fail and report that the `frontendAuthUrl` is not configured?

I'm just not sure if this fallback is correct, because I guess there are different scenarios when the frontend authentication is used comparted to the backend authentication

Reply:
This is intended behaviour. It also says in the issue: `When this property is not configured, use the current way to authenticate.`

Background is that only the pia guys want the frontend token. Climedo does not. To not break the already existing instances, SORMAS needs to default to the previous behaviour when the frontend token is not configured...

Reply:
Yep, saw that comment too, but was not sure why this was needed. I get it now, thanks